Kudu.Services/FetchHelpers/OneDriveHelper.cs

```diff
@@ -0,0 +1,519 @@
+                    }
+
+                    if (changes.ContainsKey("@changes.resync"))
+                    {
+                        if (string.IsNullOrEmpty(next))
+                        {
+                            throw new InvalidOperationException(string.Format(CultureInfo.CurrentUICulture, Resources.OneDriveUnableToSync, changes["@changes.resync"]));
+                        }
+
+                        // resync
+                        next = null;
+                        changes["@changes.hasMoreChanges"] = true;
+                        result = new ChangesResult();
+                        continue;
+                    }
+
+                    var items = serializer.ConvertToType<OneDriveModel.OneDriveItemCollection>(changes);
+
+                    // changes
+                    if (items != null && items.value != null && items.value.Length > 0)
+                    {
+                        var subResults = GetChanges(items, ids, rootUri);
+                        result.DeletionChanges.AddRange(subResults.DeletionChanges);
+                        result.DirectoryChanges.AddRange(subResults.DirectoryChanges);
+                        result.FileChanges.AddRange(subResults.FileChanges);
+                    }
+
+                    // set next token
+                    next = (string)changes["@changes.token"];
+
+                } while ((bool)changes["@changes.hasMoreChanges"]);
+
+                result.Cursor = next;
+                return result;
+            }
+        }
+
+        private ChangesResult GetChanges(OneDriveModel.OneDriveItemCollection items, Dictionary<string, OneDriveModel.ItemInfo> ids, string rootUri)
+        {
+            ChangesResult result = new ChangesResult();
+
+            foreach (var item in items.value)
+            {
+                ids[item.id] = new OneDriveModel.ItemInfo
+                {
+                    name = item.name,
+                    parentId = item.parentReference.id
+                };
+
+                var path = GetPath(ids, item);
+                if (item.deleted != null)
+                {
+                    result.DeletionChanges.Add(new OneDriveModel.OneDriveChange { Path = path, IsDeleted = true });
+                }
+                else
+                {
+                    var change = new OneDriveModel.OneDriveChange
+                    {
+                        ContentUri = string.Format(CultureInfo.InvariantCulture, "{0}/items/{1}", rootUri, item.id),
+                        Path = path,
+                        IsFile = item.file != null,
+                        LastModifiedUtc = item.lastModifiedDateTime.ToUniversalTime()
+                    };
+
+                    if (change.IsFile)
+                    {
+                        result.FileChanges.Add(change);
+                    }
+                    else
+                    {
+                        result.DirectoryChanges.Add(change);
```

Comment:
it is for updating LastWriteTimeStampUtc for folders.  see details in [line 69](https://github.com/projectkudu/kudu/pull/1579/files#diff-3c0f9e394470bf81dbb6ba78d75f86d5R69)

````
// perform action seperately, so that can ensure timestamp on directory
 // e.g two changes:
//  (new) file /a/b/c.txt
//  (new) dir  /a/b
//  if created dir first then create file. file creation will trigger folder timestamp change.
//  which will result in "/a/b" has timestamp from the monent of file creation instead of the timestamp value from server, where value supposed to be set by code specifically.
````